notes/eligibility-summary.vbs

```diff
@@ -23444,6 +23444,82 @@ class stat_detail
 		stat_mont_form_recvd_date = replace(stat_mont_form_recvd_date, " ", "/")
 		If stat_mont_form_recvd_date = "__/__/__" Then stat_mont_form_recvd_date = ""
 
+		'Herehere
+		'TODO: Is this okay placement? 
+		'TODO: Why does this only work if inside the sub?
```

Comment:
A sub (subroutine) is a function within a class and is the only place that we should be 'doing' things other than saving information. So the sub for the stat_detail class reads all of the information from stat for the specific footer month that the instance of the class is created for. 

I know that is weirdly phrased. 
A class is a structure in code to create an object with certain traits. It has information saved in variables and actions as subs. It is something that we would use over and over again that needs to function the same with different details stored within each instance. An example is in our CLoS - that is based on a class that defines everything we want to know about each script file, name, category, policy, hot topics, tags, etc. We created the initial structure and then apply that to each script. 

For Elig Summ - the stat class is the detail of STAT for any given month and we have an array of stat_detail classes from each month. The structure and type of information they contain is the same but the content of each variable within the class will be different from month to month. 

This sub is kind of a constructor sub, but we call it explicitly instead of having it happen when initiated because we set some variables and that's how I knew to do it. BUT this sub is the way we fill all the variables with the information specific to the case for the month provided.

However, if we want to read more information during the sub, we need to create variables that are defined during this run for us to read outside of the sub. At line 18613 we start by defining these variables that are specific to the class. In order to use the variables outside of the class, we need to use 'dot notation'- ClassObject.VariableName  to identify the information. We use these in an array here so it actually looks like this: STAT_INFORMATION(month_count).footer_year

So, you can move all of this outside of the sub as long as you are using the dot-notation to call the variable. Since it doesn't seem that you are reading any new information this can probably all be moved out. REVW information is read on line 23378